classification/test_sensor_data_classification.py

```diff
@@ -39,35 +92,114 @@ class TestSensorDataClassification(unittest.TestCase):
-                                                                  NUM_RECORDS,
-                                                                  PARTITIONS)
-
-      if noiseAmplitude == 0:
-        self.assertEqual(predictionAccuracy, 100)
-      else:
-        self.assertNotEqual(predictionAccuracy, 100)
+    networkConfigurations = _generateNetworkConfigurations()
+
+    for networkConfig in networkConfigurations:
+      for noiseAmplitude in WHITE_NOISE_AMPLITUDES:
+        for signalMean in SIGNAL_MEANS:
+          for signalAmplitude in SIGNAL_AMPLITUDES:
+            for signalPeriod in SIGNAL_PERIODS:
+
+              sensorType = networkConfig[SENSOR_REGION_NAME]["type"]
+              spEnabled = networkConfig[SP_REGION_NAME]["enabled"]
+              tmEnabled = networkConfig[TM_REGION_NAME]["enabled"]
+              upEnabled = networkConfig[UP_REGION_NAME]["enabled"]
+              classifierType = networkConfig[CLASSIFIER_REGION_NAME]["type"]
+
+              expParams = ("RUNNING EXPERIMENT WITH PARAMS:\n"
+                           " * numRecords=%s\n"
+                           " * signalAmplitude=%s\n"
+                           " * signalMean=%s\n"
+                           " * signalPeriod=%s\n"
+                           " * noiseAmplitude=%s\n"
+                           " * sensorType=%s\n"
+                           " * spEnabled=%s\n"
+                           " * tmEnabled=%s\n"
+                           " * upEnabled=%s\n"
+                           " * classifierType=%s\n"
+                           ) % (NUM_RECORDS,
+                                signalAmplitude,
+                                signalMean,
+                                signalPeriod,
+                                noiseAmplitude,
+                                sensorType.split(".")[1],
+                                spEnabled,
+                                tmEnabled,
+                                upEnabled,
+                                classifierType.split(".")[1])
+              print expParams
+
+              inputFile = generateData(DATA_DIR,
+                                       OUTFILE_NAME,
+                                       signalMean,
+                                       signalPeriod,
+                                       SEQUENCE_LENGTH,
+                                       NUM_RECORDS,
+                                       signalAmplitude,
+                                       NUM_CATEGORIES,
+                                       noiseAmplitude)
+
+              network = configureNetwork(inputFile,
+                                         networkConfig)
+
+              (numCorrect,
+               numTestRecords,
+               predictionAccuracy) = runNetwork(network,
+                                                networkConfig,
+                                                NUM_RECORDS)
+
+              if (noiseAmplitude == 0
```

Comment:
Would be much cleaner to have separate test methods for each of these. At the very least you should include fail messages in the assertions so we can see clearly what fails and why.